drivers/wireless/ieee80211/bcm43xxx/bcmf_driver.c

```diff
@@ -666,300 +669,247 @@ void bcmf_wl_scan_event_handler(FAR struct bcmf_dev_s *priv,
-      result_size -= BCMF_IW_EVENT_SIZE(essid)+essid_len_aligned;
-
-      /* Copy link quality info */
+      priv->scan_result_entries++;
+    }
 
-      if (result_size < BCMF_IW_EVENT_SIZE(qual))
-        {
-          goto scan_result_full;
-        }
+wl_escan_result_processed:
 
-      iwe = (struct iw_event *)&priv->scan_result[priv->scan_result_size];
-      iwe->len = BCMF_IW_EVENT_SIZE(qual);
-      iwe->cmd = IWEVQUAL;
-      iwe->u.qual.qual = bss->SNR;
-      wlinfo("signal %d %d %d\n", bss->RSSI, bss->phy_noise, bss->SNR);
-      iwe->u.qual.level = bss->RSSI;
-      iwe->u.qual.noise = bss->phy_noise;
-      iwe->u.qual.updated = IW_QUAL_DBM | IW_QUAL_ALL_UPDATED;
+  if (status == WLC_E_STATUS_PARTIAL)
+    {
+      /* More frames to come */
 
-      priv->scan_result_size += BCMF_IW_EVENT_SIZE(qual);
-      result_size -= BCMF_IW_EVENT_SIZE(qual);
+      return;
+    }
 
-      /* Copy AP mode */
+  if (status != WLC_E_STATUS_SUCCESS)
+    {
+      wlerr("Invalid event status %" PRId32 "\n", status);
+      return;
+    }
 
-      if (result_size < BCMF_IW_EVENT_SIZE(mode))
-        {
-          goto scan_result_full;
-        }
+  /* Scan done */
 
-      iwe = (struct iw_event *)&priv->scan_result[priv->scan_result_size];
-      iwe->len = BCMF_IW_EVENT_SIZE(mode);
-      iwe->cmd = SIOCGIWMODE;
-      if (bss->capability & DOT11_CAP_ESS)
-        {
-          iwe->u.mode = IW_MODE_INFRA;
-        }
-      else if (bss->capability & DOT11_CAP_IBSS)
-        {
-          iwe->u.mode = IW_MODE_ADHOC;
-        }
-      else
-        {
-          iwe->u.mode = IW_MODE_AUTO;
-        }
+  wlinfo("escan done event %" PRId32 " %" PRId32 "\n",
+         status, bcmf_getle32(&event->reason));
 
-      priv->scan_result_size += BCMF_IW_EVENT_SIZE(mode);
-      result_size -= BCMF_IW_EVENT_SIZE(mode);
+  wd_cancel(&priv->scan_timeout);
 
-      /* Copy AP encryption mode */
+  priv->scan_status = BCMF_SCAN_DONE;
+  nxsem_post(&priv->control_mutex);
 
-      if (result_size < BCMF_IW_EVENT_SIZE(data))
-        {
-          goto scan_result_full;
-        }
+  return;
 
-      iwe = (struct iw_event *)&priv->scan_result[priv->scan_result_size];
-      iwe->len = BCMF_IW_EVENT_SIZE(data);
-      iwe->cmd = SIOCGIWENCODE;
-      iwe->u.data.flags = bss->capability & DOT11_CAP_PRIVACY ?
-                          IW_ENCODE_ENABLED | IW_ENCODE_NOKEY :
-                          IW_ENCODE_DISABLED;
-      iwe->u.data.length = 0;
-      iwe->u.essid.pointer = NULL;
+exit_invalid_frame:
+  wlerr("Invalid scan result event\n");
+  bcmf_hexdump((FAR uint8_t *)event, event_len, (unsigned long)event);
+}
 
-      priv->scan_result_size += BCMF_IW_EVENT_SIZE(data);
-      result_size -= BCMF_IW_EVENT_SIZE(data);
+static FAR char *
+bcmf_wl_iwe_add_event(FAR char *stream, FAR char *stop,
+                      FAR struct iw_event *iwe, int event_len)
+{
+  if (stream + event_len > stop)
```

Comment:
why need check again? Line 833 already handle this case.

Reply:
Done